### PR TITLE
Add hyperlinks to src_loc()

### LINKED ISSUE #1396: Add links to back trace
Using `cli::style_hyperlink()` or similar.

Could also hyperlink the advice to run call `rlang::last_trace()` to save a little typing.

### PR DESCRIPTION
**Pull request by hadley**
A very quick POC.

Fixes #1396

### CODE REVIEW

File: R/trace.R [hunk 1]
```diff
@@ -884,13 +884,24 @@ src_loc <- function(srcref) {
...
+  } else {
+    file_trim <- file
   }
 
   line <- srcref[[1]]
   column <- srcref[[5]] - 1L
 
-  paste0(file, ":", line, ":", column)
+  if (is_installed("cli")) {
+    cli::style_hyperlink(
```
Comment (DavisVaughan) [inline R/trace.R:896]: Would you get weird behavior if you don't have a new enough version of cli?

Comment (hadley) [reply R/trace.R:896]: Yes, hence this being a quick POC 😄

**Comment by hadley** [comment]
Another idea: should clicking on a function name in the back trace display its source code?

**Comment by lionel-** [comment]
Nice idea.

**Comment by hadley** [comment]
Yeah, agreed. Let's leave that for now.

One other simple thing we could do in this PR would be to make `rlang::last_trace()` and `rlang::last_error()` clickable.

**Comment by hadley** [comment]
I think this is probably worth a final review and then we can merge it.

File: R/utils.R [hunk 2]
```diff
@@ -358,3 +358,10 @@ cli_with_whiteline_escapes <- function(x, fn) {
   x <- gsub("__NEW_LINE__", "\n", x, fixed = TRUE)
   x
 }
+
+style_rlang_run <- function(code) {
+  style_hyperlink(
+    paste0("`rlang::", code, "`"),
+    paste0("rstudio:run:rlang::", code)
```
Comment (lionel-) [inline R/utils.R:365]: We were discussing with @romainfrancois about making the protocol more general:

- Check if `R_CLI_HAS_HYPERLINK_RUN` envvar is set.
- If set, hyperlink to `ide:run:`

**Pull request merged**